src/sinks/kafka/service.rs

```diff
@@ -109,7 +109,14 @@ impl Service<KafkaRequest> for KafkaService {
                         event_byte_size: request.event_byte_size,
                     })
                 }
-                Err((kafka_err, _original_record)) => Err(kafka_err),
+                Err((kafka_err, _original_record)) => match kafka_err {
+                    KafkaError::MessageProduction(rdkafka::types::RDKafkaErrorCode::Fatal) => {
+                        let (_, error_description) = this.kafka_producer.client()
+                        .fatal_error().unwrap_or((rdkafka::types::RDKafkaErrorCode::Fatal, String::new()));
+                        panic!("Unrecoverable error in Kafka sink: {}, {}", kafka_err, error_description)
```

Comment:
Formatting aside, I'm not sure this is the right approach, for three reasons.
1. If `fatal_error` returns `None`, then no fatal error has occurred. It doesn't seem right to turn this non-fatal error into a fatal one.
2. Is this a system-wide fatal error, or does it just require reinitializing the consumer? If the latter, then we should ideally be looping here instead of killing Vector. At worst, this should trigger an orderly shutdown instead of a kill.
3. I don't see that this marks the events as having failed, so it could be interpreted by the source as a delivery (the default) instead as an error.